File: app/dto/currency_analytics.py
from datetime import date

from pydantic import BaseModel


class DailyChangePoint(BaseModel):
    date: date
    change: float


class DailyChangeSeries(BaseModel):
    code: str
    currency: str
    data: list[DailyChangePoint]


class MovingAveragePoint(BaseModel):
    date: date
    value: float
    ma: float | None


class TrendInfo(BaseModel):
    direction: str
    percent: float


class MovingAverageSeries(BaseModel):
    code: str
    currency: str
    data: list[MovingAveragePoint]
    trend: TrendInfo


class CorrelationResponse(BaseModel):
    codes: list[str]
    values: list[list[float]]

File: frontend/src/types/currencyAnalytics.ts
export interface DailyChangePoint {
  date: string;
  change: number;
}

export interface DailyChangeSeries {
  code: string;
  currency: string;
  data: DailyChangePoint[];
}

export interface MovingAveragePoint {
  date: string;
  value: number;
  ma: number | null;
}

export interface TrendInfo {
  direction: "up" | "down" | "flat";
  percent: number;
}

export interface MovingAverageSeries {
  code: string;
  currency: string;
  data: MovingAveragePoint[];
  trend: TrendInfo;
}

export interface CorrelationResponse {
  codes: string[];
  values: number[][];
}
